### ISSUE TITLE
First button in Dialog component is marked as :focus-visible

### ISSUE THREAD

**Issue by AndreasPresthammer** [equinor/design-system#2693]
Having a `<Button>` as the first element inside a `<Dialog.Header>` will cause the button to get the ':focus-visible' style attribute which gives it a dotted line around.

Ref slack discussion: https://equinor.slack.com/archives/CJT20H1B9/p1671020275477049

Repro:
https://codesandbox.io/s/tender-sun-2xseol

**Expected behavior**

I would expect the button not to be marked as active.

**Specifications**

See: https://codesandbox.io/s/tender-sun-2xseol
- Version: 0.27.0
- Browser: Chrome 108.0.5359.99 
- OS: Windows 11
- React 18.2.0

**Additional context**

@oddvernes mentioned in the above mentioned Slack chat that this is probably caused by the FloatingFocusManager (https://floating-ui.com/docs/floatingfocusmanager). A suggestion for a fix _might_ be to allow consumers of the <Dialog> component optionally more direct control of the props that go into the FloatingFocusManager such that we can control which element is supposed to be the 'active' one in our given context.

**Comment by oddvernes** [comment]
This is no longer relevant as we started using native dialog element.
